plugins/catalog-import/src/translation.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createTranslationRef } from '@backstage/frontend-plugin-api';
+
+/** @alpha */
+export const catalogImportTranslationRef = createTranslationRef({
+  id: 'catalog-import',
+  messages: {
+    buttons: {
+      back: 'Back',
+    },
+    defaultImportPage: {
+      headerTitle: 'Register an existing component',
+      contentHeaderTitle: 'Start tracking your component in {{appTitle}}',
+      supportTitle:
+        'Start tracking your component in {{appTitle}} by adding it to the software catalog.',
+    },
+    importInfoCard: {
+      title: 'Register an existing component',
+      deepLinkTitle: 'Learn more about the Software Catalog',
+      linkDescription:
+        'Enter the URL to your source code repository to add it to {{appTitle}}.',
+      fileLinkTitle: 'Link to an existing entity file',
+      examplePrefix: 'Example: ',
+      fileLinkDescription:
+        'The wizard analyzes the file, previews the entities, and adds them to the {{appTitle}} catalog.',
+      githubIntegration: {
+        title: 'Link to a repository',
+        label: 'GitHub only',
+        descriptionPrefix: 'The wizard discovers all ',
+        descriptionSuffix:
+          ' files in the repository, previews the entities, and adds them to the {{appTitle}} catalog.',
+        prDescriptionPrefix:
+          'If no entities are found, the wizard will prepare a Pull Request that adds an example ',
+        prDescriptionSuffix:
+          ' and prepares the {{appTitle}} catalog to load all entities as soon as the Pull Request is merged.',
```

Comment:
Try to avoid having prefix and suffix as it's very fragile and might not work well in how sentences are structured in some languages. Try to go with interpolation or worst case leave them untranslated.

Reply:
I'm not using interpolation here because the texts are used with React elements.

Ideally we should use [`Trans`](https://react.i18next.com/latest/trans-component) provided by `react-i18next`, but we are not exposing `i18next` interface in our translation plugin API, so I choose to use `prefix` and `suffix` instead temporarily, it should work for all languages although because they are in pair?

Reply:
@jhaals I tried to implement a `Translation` component at https://github.com/backstage/backstage/pull/27193/files#diff-9a9b8f6706578fbe2e5510acb56ff0b3de1767daf0bc3f2ff1c71ae86796c838R74 but `build:api-reports` runs out of memory...

Reply:
The tsc OOM disappears after upgrading i18next libs.

Reply:
@JounQin thank you! I'm thinking it's best to move this to a new PR and sort this change out by itself first?

On the direction, at a glance I'm thinking that the alternate usage of the `Trans` component might fit a bit better? https://react.i18next.com/latest/trans-component#alternative-usage-which-lists-the-components-v11.6.0. It avoids the need to list the default message multiple times.

Reply:
https://github.com/backstage/backstage/pull/27193/files#diff-1eeb293ff108472b37a75835f10938ed211d2fed61c1cbdf56d451dd19c5dcb0R298

I was using `Trans` component inside.

Reply:
Component support is added in #27193, can we migrate later?